dvc/scm/git/backend/dulwich.py

```diff
@@ -646,3 +646,17 @@ def merge(
         squash: bool = False,
     ) -> Optional[str]:
         raise NotImplementedError
+
+    def validate_git_repo(self, url: str):
+        from dulwich.client import LocalGitClient, get_transport_and_path
+        from dulwich.porcelain import get_remote_repo
+
+        try:
+            _, location = get_remote_repo(self.repo, url)
+            client, path = get_transport_and_path(location)
+        except Exception as exc:
+            raise SCMError(
+                f"'{url}' is not a valid Git remote or URL"
+            ) from exc
+        if isinstance(client, LocalGitClient) and not os.path.exists(path):
+            raise SCMError(f"'{url}' is not a valid Git remote or URL")
```

Comment:
We can make a dedicated `InvalidRemoteRepo` exception for this in `dvc.scm.base` since it seems like we may end up using it in other places as well in the future